API: Leave Group issue solving

Signior-X opened the pull request:
Fixes #18   
I have make the changes in app.py which adding the API for leaving the group. Also checked it using the postman and it is working. After deleting the data from ClubMembers table, it will return if it is success or not in JSON form.
Also added a file, to see the tables list.

Milind712000 commented on Make_Tables/mysqlconnect.py:
```diff
@@ -1,6 +1,7 @@
 import mysql.connector
 import os
 from dotenv import load_dotenv
+from mysql.connector import Error as myerror
```
- [x] Rename instances of myerror as dberror/sqlerror or something similar (make the name more informative)

Signior-X replied:
Okay, I have renamed it sqlerror

Milind712000 commented on app.py:
```diff
@@ -20,5 +21,28 @@ def Login():
     return render_template('login.html',title='Login | SAC Portal, IIT Mandi',form=form)
 
+#Example  https://host/leaveclub?userID='B19188'&clubID='C10001'
+@app.route('/leaveclub', methods = ['GET','POST'])
```
- [ ] Make this route, POST only

Signior-X replied:
It is showing method not allowed in POST, I thing it is GET, so I have make this route only GET

Milind712000 replied:
it will show method not allowed if you access it through browsers address bar because it only sends get request. that is meant to happen. that is why you should test api with postman
![Screenshot from 2020-03-31 00-43-52](https://user-images.githubusercontent.com/45260243/77952216-e77e1380-72e8-11ea-98df-14f7de4f0fc2.png)

Milind712000 replied:
Also here i have used query params but use request body to transfer data

karanankit01 replied:
You have changed the route method, but you are still using query params for data tranfer

Milind712000 commented on Make_Tables/showtables.py:
```diff
@@ -0,0 +1,13 @@
+	desription = mycursor.fetchall()
+	for j in desription:
+		print(*j)
+	print()
```
- [x] add newline at end of file
See reason : (https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file)

Milind712000 commented:
- You added a new method called get_tables_list and a file called showtables
Can you justify their need.
- You can link the issue you are working on in your PR using this https://github.blog/2013-01-22-closing-issues-via-commit-messages/ (_useful github feature_)

Signior-X commented:
Fixes #18 I made the changes in app.py as you suggested.
Sorry, I added the showtables.py by mistake, I use it on my local machine to instantly get the schema of the tables. I added it in this commit by mistake and removed it. 
I will always add a new line, learnt a new thing.
Also, that method is useless, so I deleted it.
Making the new pull request.

Signior-X committed:
Added API: Leave Group

Signior-X commented:
Also, please don't merge it now, I think merge it after the api of add groups and add members is created, so that they don't get disturbed to update their PR.

Milind712000 commented on app.py:
```diff
@@ -20,5 +21,28 @@ def Login():
+        msg = "Deleted"
+    except sqlerror as err:
+        success=0
+        msg = err
```
msg = str(err)
the err object here is not json serializable
![Screenshot from 2020-03-31 00-40-21](https://user-images.githubusercontent.com/45260243/77951810-42633b00-72e8-11ea-8918-80d359f91311.png)

Signior-X replied:
Yes, I will solve this.

Milind712000 commented:
Another thing, You don't need to commit amend or rebase everytime you make a change,
Its fine if your PR has multiple commits infact it makes it easier to see the changes that you make after each review, I'll tell you to rebase and squash once the PR is ready for merge and no more changes are required

Signior-X commented:
Okay, I will use only postman. 
Yes, I understand, what you are saying.
And one thing, thank you I am learning a lot of new things because of all this.

Signior-X committed:
Changed app.py

karanankit01 commented on app.py:
```diff
@@ -20,5 +21,28 @@ def Login():
+def leave_club():
+    data = {
+        "tablename" : "ClubMembers",
+        "userID" : request.args.get('userID'),
```
use request body to transfer data, instead of query parameters,
this'll allow for direct HTML form calls

Signior-X replied:
yes, userstood for sure, will take care from next time in PRs
However, changing it and doing it as vipul suggested, to not take any parameter, isnted use the session value.

dheeraj135 commented on app.py:
```diff
@@ -20,5 +21,28 @@ def Login():
     return render_template('login.html',title='Login | SAC Portal, IIT Mandi',form=form)
 
+#Example  https://host/leaveclub?userID='B19188'&clubID='C10001'
+@app.route('/leaveclub', methods = ['POST'])
```
Not sure about this comment. @vsvipul @Milind712000 what do you think about this?
Guard this function to check whether the user is logged in. If the user is logged in we don't need to store the userid.
If you are thinking about admin functionalities, we can have a different function for that.

Signior-X replied:
Ok, done, understood. 
I will do this.

Signior-X replied:
I will only take the clubID in the body, and take the userID from the session.

Signior-X committed:
Used the session data

Signior-X committed:
Improvements
